src/python_testing/matter_testing_infrastructure/matter/testing/matter_testing.py

```diff
@@ -222,8 +222,54 @@ def teardown_class(self):
             for problem in self.problems:
                 LOGGER.info(str(problem))
             LOGGER.info("###########################################################")
+
+            # Attempt to dump device attribute data for debugging when problems are found
+            self._dump_device_attributes_on_failure()
         super().teardown_class()
 
+    def _dump_device_attributes_on_failure(self):
+        """
+        Dump device attribute data when problems are found for debugging purposes.
+
+        This method attempts to generate a device attribute dump if the test has
+        collected endpoint data. It's designed to be safe and not interfere with
+        the original test failure reporting.
+        """
+        try:
+            # Check if we have endpoints_tlv data (from BasicCompositionTests or similar)
+            if hasattr(self, 'endpoints_tlv') and self.endpoints_tlv:
+                LOGGER.info("MatterBaseTest: Problems detected - generating device attribute dump")
+
+                # Check if we have the dump_wildcard method (from BasicCompositionTests)
+                if hasattr(self, 'dump_wildcard'):
+                    LOGGER.info("Device attribute data available - generating dump")
+                    _, txt_str = self.dump_wildcard(None)
+                    # Only dump the text format - it's more readable for debugging
+                    self._log_structured_data('==== FAILURE_DUMP_txt: ', txt_str)
+                else:
```

Comment:
I don't know if it's worthwhile logging these - it might just confuse people. Ditto with the exception.